gluten-data/src/main/scala/io/glutenproject/execution/GlutenHashAggregateExecTransformer.scala

```diff
@@ -181,28 +179,37 @@ case class GlutenHashAggregateExecTransformer(
         structTypeNodes.add(ConverterUtils.getTypeNode(LongType, nullable = true))
       case _: StddevSamp | _: StddevPop | _: VarianceSamp | _: VariancePop =>
         // Use struct type to represent Velox Row(BIGINT, DOUBLE, DOUBLE).
-        structTypeNodes.add(ConverterUtils.getTypeNode(LongType, nullable = false))
-        structTypeNodes.add(ConverterUtils.getTypeNode(DoubleType, nullable = false))
-        structTypeNodes.add(ConverterUtils.getTypeNode(DoubleType, nullable = false))
+        structTypeNodes.add(ConverterUtils
+          .getTypeNode(veloxThreeIntermediateTypes.head, nullable = false))
+        structTypeNodes.add(ConverterUtils
+          .getTypeNode(veloxThreeIntermediateTypes(1), nullable = false))
+        structTypeNodes.add(ConverterUtils
+          .getTypeNode(veloxThreeIntermediateTypes(2), nullable = false))
       case _: Corr =>
-        structTypeNodes.add(ConverterUtils.getTypeNode(DoubleType, nullable = false))
-        structTypeNodes.add(ConverterUtils.getTypeNode(LongType, nullable = false))
-        structTypeNodes.add(ConverterUtils.getTypeNode(DoubleType, nullable = false))
-        structTypeNodes.add(ConverterUtils.getTypeNode(DoubleType, nullable = false))
-        structTypeNodes.add(ConverterUtils.getTypeNode(DoubleType, nullable = false))
-        structTypeNodes.add(ConverterUtils.getTypeNode(DoubleType, nullable = false))
+        structTypeNodes.add(ConverterUtils
+          .getTypeNode(veloxSixIntermediateTypes.head, nullable = false))
+        structTypeNodes.add(ConverterUtils
+          .getTypeNode(veloxSixIntermediateTypes(1), nullable = false))
+        structTypeNodes.add(ConverterUtils
+          .getTypeNode(veloxSixIntermediateTypes(2), nullable = false))
+        structTypeNodes.add(ConverterUtils
+          .getTypeNode(veloxSixIntermediateTypes(3), nullable = false))
+        structTypeNodes.add(ConverterUtils
+          .getTypeNode(veloxSixIntermediateTypes(4), nullable = false))
+        structTypeNodes.add(ConverterUtils
+          .getTypeNode(veloxSixIntermediateTypes(5), nullable = false))
       case _: CovPopulation | _: CovSample =>
-        structTypeNodes.add(ConverterUtils.getTypeNode(DoubleType, nullable = false))
-        structTypeNodes.add(ConverterUtils.getTypeNode(LongType, nullable = false))
-        structTypeNodes.add(ConverterUtils.getTypeNode(DoubleType, nullable = false))
-        structTypeNodes.add(ConverterUtils.getTypeNode(DoubleType, nullable = false))
-        structTypeNodes.add(ConverterUtils.getTypeNode(DoubleType, nullable = true))
-        structTypeNodes.add(ConverterUtils.getTypeNode(LongType, nullable = true))
-        structTypeNodes.add(ConverterUtils.getTypeNode(DoubleType, nullable = true))
-        structTypeNodes.add(ConverterUtils.getTypeNode(DoubleType, nullable = true))
```

Comment:
Hi @liujiayi771, could you help review this change? Eight intermediate types were used here, but looks to me only four is needed.

Reply:
Only need four, It should be that a rebase has retained the wrong code.

Reply:
Got it. Thnaks for checking.